File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/client/control/camera/FirstPersonPlayerCameraTarget.java
package pize.tests.voxelgame.client.control.camera;

import pize.math.util.EulerAngles;
import pize.math.vecmath.vector.Vec3f;
import pize.tests.voxelgame.client.entity.LocalPlayer;

public class FirstPersonPlayerCameraTarget implements CameraTarget{
    
    private final LocalPlayer targetPlayer;
    private final Vec3f position;
    private final EulerAngles rotation;
    
    public FirstPersonPlayerCameraTarget(LocalPlayer targetPlayer){
        this.targetPlayer = targetPlayer;
        position = new Vec3f();
        rotation = new EulerAngles();
    }
    
    @Override
    public Vec3f getPosition(){
        position.set(targetPlayer.getLerpPosition()).add(0, targetPlayer.getEyeHeight(), 0);
        return position;
    }
    
    @Override
    public EulerAngles getRotation(){
        rotation.set(targetPlayer.getRotation());
        return rotation;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/client/control/camera/FirstPersonPlayerCameraTarget.kt
package pize.tests.voxelgame.client.control.camera

import pize.math.util.EulerAngles
import pize.math.vecmath.vector.Vec3f
import pize.tests.voxelgame.client.entity.LocalPlayer

class FirstPersonPlayerCameraTarget(private val targetPlayer: LocalPlayer?) : CameraTarget {
    private override val position: Vec3f
    private override val rotation: EulerAngles

    init {
        position = Vec3f()
        rotation = EulerAngles()
    }

    override fun getPosition(): Vec3f {
        position.set(targetPlayer.getLerpPosition()).add(0f, targetPlayer.getEyeHeight(), 0f)
        return position
    }

    override fun getRotation(): EulerAngles {
        rotation.set(targetPlayer.getRotation())
        return rotation
    }
}
